### PR TITLE
Accept MTA-STS boolean or check hostnames match

### LINKED ISSUE #146: accept MTA-STS boolean in "queue/" endpoint
this boolean being true means we should check whether they have a valid mta-sts policy, and if they do, crib all the info (mxs, enforce mode) from there, and continually validate their policy.

by default it'll be false, which means the user has to submit this info manually.

### PR DESCRIPTION
**Pull request by vbrown608**
Depends on #187 
Closes #146 

Two possible success paths:
1) they check the MTA-STS box and their most recent scan was successful with a successful MTA-STS check
    - In this case we can use the `PreferredHostnames` from their scan, which must be a subset of the hostnames listed in their MTA-STS policy file.
2) they provide hostnames, their most recent scan was successful, and the hostnames they provided are a subset of the `PreferredHostnames` in their scan.

### CODE REVIEW

File: models/domain.go [hunk 1]
```diff
@@ -35,20 +36,55 @@ type scanStore interface {
...
+func (d *Domain) PopulateFromScan(scan Scan) {
+	// We should only trust MTA-STS info from a successful MTA-STS check.
+	if scan.Data.MTASTSResult != nil && scan.SupportsMTASTS() {
+		d.MTASTSMode = scan.Data.MTASTSResult.Mode
+		// If the domain's MXs are missing, we can take them from the scan's
+		// PreferredHostnames, which must be a subset of those listed in the
+		// MTA-STS policy file.
+		if len(d.MXs) == 0 {
+			d.MXs = scan.Data.PreferredHostnames
+		}
```
Comment (sydneyli) [inline models/domain.go]: We should populate this from the actual MTA-STS policy file result!

Comment (vbrown608) [reply models/domain.go]: Okay cool. My understanding was that "preferred hostnames" are all the hostnames listened in their DNS record that we were able to connect to, and that the hostnames listed in a valid MTA-STS record would be a superset of those. Is the idea that we want to capture hosts in the MTA-STS record that they might want to use in the future?

File: models/domain.go [hunk 2]
```diff
@@ -35,20 +36,55 @@ type scanStore interface {
...
 	}
-	// Check to see it's not already on the Policy List.
 	if list.HasDomain(d.Name) {
-		return false, "Domain is already on the policy list!"
+		return false, "Domain is already on the policy list!", scan
 	}
-	return true, ""
+	if d.MTASTSMode != "" && !scan.SupportsMTASTS() {
+		return false, "Domain does not correctly implement MTA-STS.", scan
+	} else if !subset(d.MXs, scan.Data.PreferredHostnames) {
```
[the remark this answers is not in the document]
Comment (vbrown608) [reply models/domain.go]: Oh yeah good point!

File: models/domain.go (hunk 1 above)
Comment (vbrown608) [reply models/domain.go]: Oh but I guess the policy is a little different from a list of actual hostnames...

**Pull request merged**